examples/slave/nuttx/xmc4800/EmulatedEEPROM.cc

```diff
@@ -0,0 +1,203 @@
+
+                // ACK command, error bit cleared by writing command, clear CRC errors, all other bits read only
+                eeprom::Command write = eeprom::Command::WRITE;
+                memcpy((uint16_t *)ECAT_EEP_CONT_STAT, &write, sizeof(control_status_reg));
+
+                shall_write_ = true;
+                last_write_ = duration_cast<milliseconds>(since_epoch());
+                break;
+            }
+
+            case eeprom::Command::RELOAD:
+            {
+                eeprom_description eeprom_desc;
+                std::memcpy(&eeprom_desc, eeprom_.data, sizeof(eeprom_description));
+
+                // INFINEON way of implementing the reload, seems to differ from EtherCAT register description section II,
+                // Table 67: Register EEPROM Data for EEPROM Emulation Reload IP Core (0x0508:0x050F)
+                uint32_t data_to_load = eeprom_desc.pdi_control + (eeprom_desc.pdi_configuration << 16);
+                memcpy((uint32_t *)ECAT_EEP_DATA, &data_to_load, sizeof(data_to_load));
+                data_to_load = eeprom_desc.sync_impulse_length + (eeprom_desc.pdi_configuration_2 << 16);
+                memcpy((uint32_t *)(ECAT_EEP_DATA + 4), &data_to_load, sizeof(data_to_load));
+                eeprom::Command reload = eeprom::Command::RELOAD;
+                memcpy((uint16_t *)ECAT_EEP_CONT_STAT, &reload, sizeof(eeprom::Command));
+
+                data_to_load = eeprom_desc.station_alias + (eeprom_desc.reserved1 << 16);
+                memcpy((uint32_t *)ECAT_EEP_DATA, &data_to_load, sizeof(data_to_load));
+                data_to_load = eeprom_desc.reserved2 + (eeprom_desc.crc << 16);
+                memcpy((uint32_t *)(ECAT_EEP_DATA + 4), &data_to_load, sizeof(data_to_load));
+                memcpy((uint16_t *)ECAT_EEP_CONT_STAT, &reload, sizeof(eeprom::Command));
+
+                // wait until eeprom is loaded.
+                while (*(uint16_t *)(ECAT_AL_EVENT_REQUEST) & EEPROM_LOADING_STATUS);
+
+                _info("eeprom RELOAD command\n");
+                break;
+            }
+            case eeprom::Command::NOP:
+            {
+                break;
+            }
+            default:
+            {
+                _info("Invalid eeprom command %x \n", control_status_reg);
+            }
+        }
+    }
+
+    void EmulatedEEPROM::load_from_flash()
+    {
+        memcpy(&eeprom_, (uint16_t*) EEPROM_FLASH_ADR, sizeof(eeprom_));
+
+        uint32_t adler_sum = adler32Sum(eeprom_.data, EEPROM_MAX_SIZE_BYTE);
+
+        _info("Adler sum read %lx, computed %lx\n", eeprom_.adler_checksum, adler_sum);
+
+        if (adler_sum != eeprom_.adler_checksum)
+        {
+            _info("Checksum Mismatch !! Load default eeprom \n");
+            load_default_eeprom_ = true;
+        }
+
+        // debug print:
+        for(uint32_t i = 0; i < 20; i++)
+        {
+            _info("EEPROM %li : %x\n", i, eeprom_.data[i]);
```

Comment:
prefer the version that can be disabled (slave_info? maybe add sii_info?)

Reply:
Nice, I missed the existence of Kickcat/debug and was using the _info from NuttX, switch to kickcat's one :)